tools/plist_to_html/plist_to_html/PlistToHtml.py

```diff
@@ -169,45 +158,59 @@ def create_index_html(self, output_dir):
         report_data = sorted(report_data,
                              key=lambda d: d['report']['path'])
 
-        # Create table lines.
-        for i, data in enumerate(report_data):
-            html_file = data['html_file']
-            report = data['report']
-
-            events = report['events']
-            checker = report['checkerName']
-            severity = report['severity']
-
-            review_status = report['reviewStatus'] \
-                if 'reviewStatus' in report and report['reviewStatus'] else ''
-
-            table_reports += '''
-              <tr>
-                <td>{0}</td>
-                <td file="{3}" line="{4}">
-                  <a href="{1}#reportHash={2}">{3} @ Line {4}</a>
-                </td>
-                <td class="severity" severity="{5}">
-                  <i class="severity-{5}"></i>
-                </td>
-                <td>{6}</td>
-                <td>{7}</td>
-                <td class="bug-path-length">{8}</td>
-                <td class="review-status review-status-{9}">{10}</td>
-              </tr>'''.format(i + 1,
-                              os.path.basename(html_file),
-                              report['reportHash'],
-                              report['path'],
-                              events[-1]['location']['line'],
-                              severity.lower(),
-                              checker,
-                              events[-1]['message'],
-                              len(events),
-                              review_status.lower().replace(' ', '-'),
-                              review_status)
-
-        substitute_data = self._tag_contents
-        substitute_data.update({'table_reports': table_reports})
+        with io.StringIO() as table_reports:
```

Comment:
Wow! Is this like using `std::ostringstream` to buffer or format an output?

Reply:
Yes, it is.